app/controllers/CampaignApi.scala

```diff
@@ -105,8 +109,15 @@ class CampaignApi(components: ControllerComponents, authAction: AuthAction[AnyCo
     }
   }
 
-  def getCampaignReferrals(campaignId: String) = authAction { _ =>
-    CampaignReferralRepository.getCampaignReferrals(campaignId) match {
+  def getCampaignReferrals(campaignId: String, start: Option[String], end: Option[String]) = authAction { _ =>
+    def toDate(o: Option[String]): Option[LocalDate] = o flatMap { s =>
+      Try(LocalDate.parse(s)).toOption
+    }
+    val dateRange = for {
+      from <- toDate(start)
```

Comment:
What happens if a date range is not specified? Will it show all in this case as expected?

Reply:
The idea is `dateRange` will then be `None`, which will mean the filter won't apply in `CampaignReferralRepository`, so then all the campaign figures will be fetched.

Reply:
So you have to give both dates if you want filtered results, which seems like a reasonable first iteration.